src/registry.ts

```diff
@@ -0,0 +1,41 @@
+import { Document } from "@govtechsg/open-attestation";
+import { ethers, Signer } from "ethers";
+import { getBatchMerkleRoot } from "./util/token";
+import { abi as TokenRegistryABI } from "../build/contracts/ERC721MintableFull.json";
+
+export class TokenRegistry {
+  web3Provider: Signer;
+
+  address: string;
+
+  contractInstance: any;
+
+  /**
+   * Creates a TokenRegistry instance with the specified address and ethersjs signer
+   * @param {object} instanceParameters
+   * @param {string} instanceParameters.contractAddress - Address that the TokenRegistry is located at
+   * @param {Signer} instanceParameters.web3Provider - An ethers.js signer instance with read/write access
+   */
+  constructor({ contractAddress, web3Provider }: { contractAddress: string; web3Provider: Signer }) {
+    this.web3Provider = web3Provider;
+    this.address = contractAddress;
+    // doing JSON.stringify below because of ethers.js type issue: https://github.com/ethers-io/ethers.js/issues/602
+    this.contractInstance = new ethers.Contract(contractAddress, JSON.stringify(TokenRegistryABI), web3Provider);
+  }
+
+  async mint(document: Document, owner: string) {
```

Comment:
- do you mind trying to upgrade to open-attestation v3
- isn't it a wrapped document ?

Reply:
must I upgrade? can I wait until you sort out v3? :D
it is a wrapped document indeed

Reply:
it might be worth to still upgrade to open-attestation v3 as the typing has been improved, it doesnt mean you need to handle all document you can only focus on v2 document: `WrappedDocument<v2.OpenAttestationDocument>`. We can eventually go through this together

Dont forget to use getData if you handle WrappedDocument

Reply:
i don't need to use getData because i'm using the content of the wrapper

Reply:
:ok: